### ISSUE TITLE
GraphQL nearObject distance variable does not accept float

### ISSUE THREAD

**Issue by adlerfaulkner** [semi-technologies/weaviate#2337]
When querying with a `nearObject` filter with a `distance` query variable that is a float, I get an error saying `Variable \"$distance\" got invalid value 0.174.\nExpected type \"Float\", found \"0.174\".` This seems like a bug because I am supplying a float in the query variables:

<img width="1049" alt="image" src="https://user-images.githubusercontent.com/8572428/199310665-95cb3143-68f9-42fe-b496-cde3ad636921.png">

However, when I use a float directly in the GraphQL query, I do not get the error. (See correct result on right of image below)
<img width="1145" alt="CleanShot 2022-11-01 at 18 29 25@2x" src="https://user-images.githubusercontent.com/8572428/199310793-45780dc4-f44a-4603-bda7-519f55c21943.png">

And when I change the query variable to a string, I do not get the error, though the Weaviate console's linter gives a warning. (See correct result on right of image below)
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/8572428/199310580-9f0a9018-c299-4485-8ad6-9fc6306f191a.png">

I have replicated all three cases above using the GraphQL API, so it's not just a bug in the Weaviate Cloud Console.

**Comment by trengrj** [comment]
Thanks @adlerfaulkner for raising this, I can reproduce using curl.

```graphql
query($distance: Float){
    Get {
        Article(nearObject : {
            id: "51b3b5c0-39b8-4a4d-a05a-0a621d326481"
            distance: $distance
        }) {
            _additional {
                id
            }
        }
    }
}
```

```bash
curl --location --request POST 'http://localhost:8080/v1/graphql' \ --header 'Content-Type: application/json' --data-raw '{"query":"query($distance: Float){\n    Get {\n        Article(nearObject : {\n            id: \"51b3b5c0-39b8-4a4d-a05a-0a621d326481\"\n            distance: $distance\n        }) {\n            _additional {\n                id\n            }\n        }\n    }\n}","variables":{"distance":1.2}}'
```

**Comment by trengrj** [comment]
This looks fixed now in 1.16.5 due to the recent upgrade of `graphql-go`.

<img width="1325" alt="Screen Shot 2022-11-25 at 4 46 02 pm" src="https://user-images.githubusercontent.com/2173919/203917523-8132bf4b-497c-4061-941a-c76870fbcd8c.png">
